### PR TITLE
Add Dependabot config

### PR DESCRIPTION
**Pull request by tony-iqlusion**
Automatically opens PRs to update out-of-date dependencies:

https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/configuring-dependabot-security-updates

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+- package-ecosystem: cargo
+  directory: "/"
+  schedule:
+    interval: weekly
```
Comment (tony-iqlusion) [inline .github/dependabot.yml:6]: `weekly` helps reduce overall churn, especially for dependencies that release frequently

**Comment by jkilpatr** [comment]
In my experience dependabot opens a lot of random PR's especially in projects with lots of dependencies. 

Why do we need dependabot versus just not specifying the patch version in our dependencies list? With the exclusion of patch versions most things don't get updated but every several months.

**Comment by tony-iqlusion** [comment]
Since Cargo.lock is checked in, Dependabot also ensures we have the latest versions in Cargo.lock.

We can turn down the settings to reduce the overall number of PRs, for example reducing the number of concurrently open PRs.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  open-pull-requests-limit: 10
```
Comment (tony-iqlusion) [inline .github/dependabot.yml:7]: This can be adjusted down if desired.

**Comment by jkilpatr** [comment]
> Since Cargo.lock is checked in, Dependabot also ensures we have the latest versions in Cargo.lock.

Wouldn't this technically require opening a pull request any time any transitive dependency changed? I don't think this is feasible just from a sheer scale perspective, you have update a lot of transitive deps at once otherwise we'll be looking at hundreds of PRs.

File: .github/dependabot.yml (hunk 2 above)
Comment (jkilpatr) [reply .github/dependabot.yml:7]: I think 5 is good. Keep things on major versions only and don't automatically update the cargo.lock, there's no substitute for knowing what deps tree everyone is using in production. If you don't keep track you'll at some point regret it.

**Comment by tony-iqlusion** [comment]
I believe Dependabot will only upgrade immediate dependencies as specified in `Cargo.toml`, but does the equivalent of `cargo update -p <dependency>`, so it will upgrade transitive dependencies indirectly as part of the PRs it opens.

**Review by jkilpatr — APPROVED** [review]

**Pull request merged**